acm/src/main/java/org/literacybridge/acm/config/DBConfiguration.java

```diff
@@ -546,6 +540,23 @@ public void setFuzzyThreshold(int threshold) {
         getDbProperties().setProperty(Constants.FUZZY_THRESHOLD, Integer.toString(threshold));
     }
 
+    public Integer getVolumeStep() {
+        Integer result = null;
```

Comment:
It is unclear, reading this function, why you return an Integer rather than an int. The usual reason is to distinguish between "no value" and a valid integer value (which is how this function is used, but you don't document that in the function header). Here, you always return either Constants.VOLUME_STEP_DEFAULT or whatever the property parses to.

So the function can never return null, and addSystemDirFilesToImage will ALWAYS add volume_step to a config.txt file.

To fix, remove line 546, "result = Constants.VOLUME_STEP_DEFAULT;"

Also, add proper Javadoc to the function. You should ALWAYS add accurate Javadoc for every function you ever write or even edit. I find that it helps me avoid errors like this. ALWAYS. EVERY TIME. NO EXCEPTIONS. I don't care how annoying or trivial, you need the habit. We all do. (Myself included. If you ever review my code, and I didn't write a function header, you should yell at me!)